funcx_endpoint/funcx_endpoint/endpoint/rabbit_mq/result_queue_publisher.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import time
+
+import pika
+
+logger = logging.getLogger(__name__)
+
+
+class ResultQueuePublisher:
+    """ResultPublisher publishes results to a topic exchange, with the {endpoint_id}.results
+    as a routing key.
+    """
+
+    def __init__(
+        self,
+        endpoint_id: str,
+        pika_conn_params: pika.ConnectionParameters,
+        exchange="results",
+    ):
+        self.endpoint_id = endpoint_id
+        self.routing_key = f"{self.endpoint_id}.results"
+        self.params = pika_conn_params
+        self.params.heartbeat = 30
+        self.params.blocked_connection_timeout = 60
+        self.exchange = exchange
+        self.exchange_type = "topic"
+        self._is_connected = False
```

Comment:
I don't see `_is_connected` being used anywhere.
Could we instead use

```python
@property
def is_connected(self) -> bool:
    return self._channel is not None
```
?

The fewer explicit attributes we're trying to track and keep in sync, the less chance we have of subtle bugs from them getting out of sync.